### ISSUE TITLE
Windows meson build has undefined variable problems

### ISSUE THREAD

**Issue by talisein** [libxmlplusplus/libxmlplusplus#67]
https://github.com/libxmlplusplus/libxmlplusplus/blob/ab5098e2a8b7fe69a991a28fe00a75e078f9a90e/meson.build#L110-L117

The `required:` is problematic, because projects using libxml++ may also depend on libxml-2.0 themselves. That means meson will resolve the dep and cache it before it sees this Requires. So xml2_dep.found() will be true on windows. This means...

https://github.com/libxmlplusplus/libxmlplusplus/blob/ab5098e2a8b7fe69a991a28fe00a75e078f9a90e/meson.build#L131

This `winsock_dep`, `bcrypt_dep`, and `xml2_sp` variables which are inside the `if not xml2_dep.found()` won't be set. `xml2_sp` is only used if `build-tests` is true, but there's no way around needing the winsock_dep variable.

**Comment by kjellahl** [comment]
@talisein @fanc999-1 You two made it possible to build libxml2 as a subproject
in Windows. Can you fix the problem with undefined variables?

As an aside: libxml2 can now be built with Meson. That would simplify, wouldn't it?
The meson.build files in libxml2 are new (about 2 months). They are not yet
included in a released version of libxml2. Might be too soon to start depending on
them now. I don't know if the meson build is still considered experimental.

**Comment by talisein** [comment]
I'll give it a shot this week

**Comment by talisein** [comment]
I think the easiest thing to do is wait for this PR to get merged, then switch to using it as a normal meson subproject

https://gitlab.gnome.org/GNOME/libxml2/-/merge_requests/244

**Comment by jamesrwaugh** [comment]
Just ran into this, I suppose a new release of libxmlplusplus is needed and a new wrap merged into Meson.
Not just on Window, but Linux as well.

**Comment by talisein** [comment]
What are the details of your Linux failure? iirc only that winsock stuff would be relevant

**Comment by jamesrwaugh** [comment]
@talisein 

The issue occurs when trying to add libxml2 as a wrap. Issue does not occur when it is system-installed, since the faulty line is only when installed as sub-project (`xml2_is_subproject == true`)

Relevant Meson lines:

```meson
subproject('libxml2')
subproject('libxmlpp')

libxmlpp = dependency('libxml++')

executable(
    'corescanner',
    src,
    dependencies: [
        libxmlpp
    ],
)
```

When building:

```
Executing subproject libxmlpp 

libxmlpp| Project name: libxml++
libxmlpp| Project version: 5.2.0
libxmlpp| C++ compiler for the host machine: c++ (gcc 13.2.0 "c++ (Ubuntu 13.2.0-23ubuntu4) 13.2.0")
libxmlpp| C++ linker for the host machine: c++ ld.bfd 2.42
libxmlpp| Program python3 found: YES 3.12.3 3.12.3 (/usr/bin/python3)
libxmlpp| Dependency libxml-2.0 found: YES 2.13.3 (overridden)

../subprojects/libxml++-5.2.0/meson.build:166:27: ERROR: Unknown variable "xml2_sp".
```

In the file we see that `xml2_sp` is filled out in a scope above, so it does not exist on that line. It does not seem to have to do with winsock. But this all seems to have been resolved in the latest master branch.

**Comment by talisein** [comment]
I see. Yes, it looks like #68 also solves that.

**Comment by kjellahl** [comment]
I have released libxml++ 5.2.1.
